filters/CropFilter.cpp

```diff
@@ -257,6 +275,29 @@ bool CropFilter::crop(const PointRef& point, const BOX2D& box)
     return (m_args->m_cropOutside != box.contains(x, y));
 }
 
+void CropFilter::crop(const Bounds& box, PointView& input, PointView& output)
```

Comment:
This doesn't seem to be used.

Reply:
It is used https://github.com/PDAL/PDAL/blob/ed3c8e5035574cc723e88a697756f5e18757315e/filters/CropFilter.cpp#L245

Reply:
OK.  Then I think the BOX2D version isn't used.

I think this logic would be better in the caller.  is3D is invariant.  No need to check for every point.  The compiler might optimize this, but IMO would still be cleaner to eliminate the function altogether and do:

```
if (box.is3d())
  crop(box.to3d(), *view, *outView);
else
  crop(box.to2d(), *view, *outView);
```

Or eliminate the BOX2D version.  Do what you want.